### PR TITLE
Add ephemeral storage size option

### PR DESCRIPTION
**Pull request by drselump14**
# What 

* Adds an option to set ephemeral storage size via a new variable: `ephemeral_storage_size`
* Removes variables that were not used in the code:
	* `event_source_mappings`
	* `ignore_external_function_updates`
	* `sns_subscriptions`
	* `cloudwatch_event_rules`
	* `cloudwatch_log_subscription_filters`

# Why

* Enables module consumers to set larger ephemeral disk space for their Lambdas
* Cleans up unused code.

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -55,6 +55,12 @@ variable "lambda_environment" {
   default     = null
 }
 
+variable "ephemeral_storage_size" {
+  type        = number
+  description = "The amount of storage in GB to allocate for the Lambda function's file system."
```
Comment (Gowiem) [inline variables.tf:60]:
```suggestion
  description = <<EOF
  The size of the Lambda function Ephemeral storage (/tmp) represented in MB. 
  The minimum supported ephemeral_storage value defaults to 512MB and the maximum supported value is 10240MB.
  EOF
```

**Comment by max-lobur** [comment]
> /test all

@Gowiem `/terratest` is the new command, it does only terratest, everything else is automatic per-commit

**Comment by drselump14** [comment]
done!

**Comment by drselump14** [comment]
> > done!
> 
> Thanks! tflint has found that variable `event_source_mappings` is unused, would you mind removing it? Also when you done please run the
> 
> ```
>   make init
>   make readme
> ```
> 
>     * this will get your docs up to date

Done. I removed other variables that is no longer used too

**Comment by max-lobur** [comment]
Thanks for the cleanup!

**Comment by manifestori** [comment]
This looks great, whats holding us back?

**Comment by manifestori** [comment]
`event_source_mappings` was supposed to be used; why remove it? how can one pass triggers to lambda...

**Comment by drselump14** [comment]
> `event_source_mappings` was supposed to be used; why remove it? how can one pass triggers to lambda...

Hi, not sure buf tflint complaints about the variable not being used, and I can't find any reference to this variable outside the variable.tf file

**Review by Gowiem — APPROVED** [review]

**Pull request merged**